clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10429,6 +10429,8 @@ def err_conflicting_sycl_kernel_attributes : Error<
   "conflicting attributes applied to a SYCL kernel">;
 def err_conflicting_sycl_function_attributes : Error<
    "%0 attribute conflicts with '%1' attribute">;
+def err_sycl_typecheck_zero_array_size : Error<
+  "zero-length arrays are not permitted in SYCL kernels">;
```

Comment:
I think we can use `err_typecheck_zero_array_size` diagnostics.

Reply:
If we do, that will output the message "zero-length arrays are not permitted in C++" when a zero length array is encountered from a SYCL kernel ( * ).  Is that ok? 

I think the "zero-length arrays are not permitted in SYCL kernels" is more accurate.  But the decision is yours.

( * ) - Zero length arrays elsewhere will still be accepted ( and I don't think we should change that behavior).

Reply:
If you want customize the message, you can make this diagnostics parameterized and emit different wording for SYCL. 
I'm fine with the original message.